types/eslint/index.d.ts

```diff
@@ -1373,13 +1392,45 @@ export namespace ESLint {
     }
 
     interface Plugin extends ObjectMetaProperties {
-        configs?: Record<string, ConfigData | Linter.FlatConfig | Linter.FlatConfig[]> | undefined;
+        configs?: Record<string, ConfigData | Linter.Config | Linter.Config[]> | undefined;
```

Comment:
Should we also allow `LegacyConfig` here?

Reply:
It looks like `ConfigData` is the same as `Linter.LegacyConfig` except for omitting the `$schema` property:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7bbd183e29bffb937b00cc115af691c9da0a3b1a/types/eslint/index.d.ts#L1351

I'm not sure why `$schema` was declared in the first place since ESLint doesn't use it. Probably we could just replace `ConfigData` with `Linter.LegacyConfig` here.